src/Validator/EditorValidator.sol

```diff
@@ -3,33 +3,52 @@ pragma solidity ^0.8.13;
 
 import {IValidator} from "./IValidator.sol";
 
-// set this to wiki as validator and it will check if user has editor NFT and has not more than 5 edits in last 24h . number of edits is settable
 contract EditorValidator is IValidator {
-
     /// -----------------------------------------------------------------------
     /// Errors
     /// -----------------------------------------------------------------------
 
     error WrongIPFSLength();
+    error ExceededEditLimit();
 
     /// -----------------------------------------------------------------------
     /// Storage variables
     /// -----------------------------------------------------------------------
-
-    /// @notice The amount of edits
-    mapping(address => uint256) public edits;
+    mapping(address => uint256[]) edits;
+    uint256 count;
 
     /// -----------------------------------------------------------------------
     /// External functions
     /// -----------------------------------------------------------------------
 
     // mapping of editors and edits / blocktimestamps in order of calculate how many edits in X period of time
-    function validate(address _user, string calldata _ipfs) external returns (bool) {
-        edits[_user]++; // TODO: dummy code
+    function validate(address _user, string calldata _ipfs)
+        external
+        returns (bool)
+    {
+        // check if array is full
+        if (edits[_user].length < 5) {
+            edits[_user].push(uint256(block.timestamp)); // add new element to array
+        } else if (edits[_user].length == 5) // check if array is full

```

Comment:
no need of else if and checking storage variable which its expensive in gas. If its not less than 5 its obv == since you wont allow them to create more than 5